crm_lead_to_event_registration/models/crm_lead.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    event_registration_id = fields.Many2one(
+        "event.registration",
+        "Event registration",
+        readonly=True,
+        help="Registration generated by this lead/opportunity.")
+
+    @api.multi
+    def action_generate_event_registration(self, event):
+        """Generate an event registration."""
+        er = self.env["event.registration"]
+        for s in self:
+            s.event_registration_id = er.create({
+                "event_id": event.id,
+                "partner_id": s.partner_id.id,
+                "nb_register": 1,
+                "name": s.contact_name,
```

Comment:
Registration could fallback to ```s.partner_name``` and then ```s.partner_id.name```. If none defined a ValidationError could be raised because in that case registration name would be 'False' as you can see in the snapshot

![image](https://cloud.githubusercontent.com/assets/1881111/13150597/b8f88a18-d666-11e5-92da-c0c2d150049d.png)